### PR TITLE
Wip master

### PR DESCRIPTION
**Pull request by alexanderGugel**
Seems kind of weird to do this without a dispatcher though...
The store tests are still the same... maybe a bit too short.

Not sure yet how to best test the actions in this case without mocking out jsonp...

Thoughts?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -7,13 +7,94 @@
...
+  },
+  {
+    "title": "search result 3",
+    "description": "longer description of search result 3",
+    "link": "https://en.wikipedia.org/wiki/search_result_3"
+  }
+]
+```
+
+2. The last time the store was updated.
```
Comment (thomastuts) [inline README.md:76]: This numbering is messed up (displays as `1. The last...`)

File: package.json [hunk 2]
```diff
@@ -45,6 +46,7 @@
     "react-addons-test-utils": "^15.3.0"
   },
   "dependencies": {
+    "jsonp": "^0.2.0",
```
Comment (thomastuts) [inline package.json:51]: Needs `sinon` as a dependency

**Review by thomastuts — CHANGES_REQUESTED** [review]
@alexanderGugel Great stuff yet again! I like how we also transform the data instead of using it as-is — makes students think about data structures. Good job! 🔥

**Comment by thomastuts** [comment]
For testing the XHR stuff, you can use `fetch` instead and use `fetch-mock` like in [this lab](https://github.com/learn-co-curriculum/react-props-and-state-lab/blob/master/fetch-setup.js).

**Comment by alexanderGugel** [comment]
@thomastuts it looks like we have to use JSONP. The API doesn't allow CORS (https://phabricator.wikimedia.org/T62835).

**Comment by alexanderGugel** [comment]
@thomastuts In order to make things testable, I extracted out the "minimal" possible Wikipedia API. This makes things easier to test without hacking around `require` and also makes it easier for students to concentrate on the actual action creators — instead of fighting with CORS issues etc.

Can I merge?

**Comment by thomastuts** [comment]
@alexanderGugel This looks great! Good solution for the XHR stuff, too. Nice job! 💣

**Pull request merged**